src/pages/[platform]/build-a-backend/server-side-rendering/nuxt/index.mdx

```diff
@@ -0,0 +1,626 @@
+
+</Callout>
+
+### Implement `02.auth-redirect.ts`
+
+Example implementation:
+
+```ts title="plugins/02.auth-redirect.ts"
+import { Amplify } from 'aws-amplify';
+import config from '~/amplify_outputs.json';
+
+// Amplify.configure() only needs to be called on the client side
+if (process.client) {
+  Amplify.configure(config, { ssr: true });
+}
+
+export default defineNuxtPlugin({
+  name: 'AmplifyAuthRedirect',
+  enforce: 'pre',
+  setup() {
+    addRouteMiddleware(
+      'AmplifyAuthMiddleware',
+      defineNuxtRouteMiddleware(async (to) => {
+        try {
+          const session = await useNuxtApp().$Amplify.Auth.fetchAuthSession();
+
+          // If the request is not associated with a valid user session
+          // redirect to the `/sign-in` route.
+          // You can also add route match rules against `to.path`
+          if (session.tokens === undefined && to.path !== '/sign-in') {
+            return navigateTo('/sign-in');
+          }
+
+          if (session.tokens !== undefined && to.path === '/sign-in') {
+            return navigateTo('/');
+          }
+        } catch (e) {
+          if (to.path !== '/sign-in') {
+            return navigateTo('/sign-in');
+          }
+        }
+      }),
+      { global: true }
+    );
+  }
+});
+```
+
+<Callout warning="true">
+
+Make sure you call `Amplify.configure` as early as possible in your application’s life-cycle. A missing configuration or `NoCredentials` error is thrown if `Amplify.configure` has not been called before other Amplify JavaScript APIs. Review the [Library Not Configured Troubleshooting guide](/gen1/[platform]/build-a-backend/troubleshooting/library-not-configured/) for possible causes of this issue.
+
+</Callout>
+
+## Set Up Amplify for API Route Use Cases
+
+Following the specification of Nuxt, your API route handlers will live under `~/server`, which is a separate environment from other parts of your Nuxt app; hence, the plugins created in the previous sections are not usable here, and extra work is required.
+
+### Set Up Amplify Server Context Utility
+
+1. If you haven’t already done so, create a `utils` directory under the server directory of your Nuxt project
+2. Create an `amplifyUtils.ts` file under the `utils` directory
+
+In this file, you will create a helper function to call Amplify APIs that are capable of running on the server side with context isolation.
+
+Example implementation:
+
+
+```ts title="utils/amplifyUtils.ts"
+import type { H3Event, EventHandlerRequest } from 'h3';
+import {
+  createKeyValueStorageFromCookieStorageAdapter,
+  createUserPoolsTokenProvider,
+  createAWSCredentialsAndIdentityIdProvider,
+  runWithAmplifyServerContext,
+  AmplifyServer,
+  CookieStorage
+} from 'aws-amplify/adapter-core';
+import { parseAmplifyConfig } from 'aws-amplify/utils';
+
+import type { LibraryOptions } from '@aws-amplify/core';
```

Comment:
same nit for grouping type imports at the top of the file. Additionally, does this break when using npm's `install-strategy=linked`, yarn, or pnpm? pnpm will not resolve imports from transient deps

Reply:
Tested with `install-strategy=linked` Nuxt itself started failing while running `postinstall` script. Consider the following statement printed by npm

```
npm WARN reify The "linked" install strategy is EXPERIMENTAL and may contain bugs.
```

I think we can use this import path as is for now.

Reply:
is this type export available from the top-level package? pnpm is unable to resolve it
![CleanShot 2024-05-16 at 13 02 33@2x](https://github.com/aws-amplify/docs/assets/5033303/1e0d5135-dc09-4e99-9cec-b5f5fb7c99cc)